cantools/database/can/signal.py

```diff
@@ -305,6 +321,41 @@ def __init__(self,
             # multi-lingual dictionary
             self.comments = comment
 
+    @property
+    def initial(self) -> Optional[Union[int, SignalValueType]]:
+        warnings.warn(
+            "The `Signal.initial` attribute is deprecated and scheduled for removal in April 2024. "
+            "Use `Signal.raw_initial` or `Signal.scaled_initial` instead.",
+            DeprecationWarning,
+        )
+        return self._initial
+
+    @initial.setter
+    def initial(self, value: Union[int, SignalValueType]) -> None:
+        warnings.warn(
+            "The `Signal.initial` attribute is deprecated and scheduled for removal in April 2024. "
+            "Use `Signal.raw_initial` or `Signal.scaled_initial` instead.",
+            DeprecationWarning,
+        )
+        self._initial = value
+
+    def raw_to_scaled(self, raw: int) -> SignalValueType:
+        with contextlib.suppress(KeyError, TypeError):
+            return self.choices[raw]  # type: ignore[index]
+
+        return raw * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled: SignalValueType) -> int:
+        if isinstance(scaled, (float, int)):
+            _transform = float if self.is_float else round
+            if self.offset == 0 and self.scale == 1:
+                # treat special case to avoid introduction of unnecessary rounding error
+                return _transform(scaled)  # type: ignore[operator,no-any-return]
```

Comment:
once the return type is adapted above, the `type: ignore` should become unnecessary...

Reply:
mypy doesn't know the return value of `_transform`, so it will complain here